moduleframework/avocado_testers/avocado_test.py

```diff
@@ -263,6 +263,19 @@ def run_script(self, *args, **kwargs):
         """
         return self.backend.run_script(*args, **kwargs)
 
+    def mark_as_warn(self, func, *args, **kwargs):
```

Comment:
maybe would be better call this functin, `assert_to_warn` to explicitly say, that you are transforming assert, not any exception. I know that I've also choosed this name, I were lazy to invent some better one :-)